pkg/operation/botanist/component/vpnseedserver/vpn_seed_server_test.go

```diff
@@ -0,0 +1,1003 @@
+										{
+											Name:      DeploymentName,
+											MountPath: "/srv/secrets/vpn-server",
+										},
+										{
+											Name:      VpnSeedServerTLSAuth,
+											MountPath: "/srv/secrets/tlsauth",
+										},
+										{
+											Name:      "vpn-seed-server-dh",
+											MountPath: "/srv/secrets/dh",
+										},
+									},
+								},
+								{
+									Name:            "envoy-proxy",
+									Image:           envoyImage,
+									ImagePullPolicy: corev1.PullIfNotPresent,
+									SecurityContext: &corev1.SecurityContext{
+										Capabilities: &corev1.Capabilities{
+											Add: []corev1.Capability{
+												"NET_BIND_SERVICE",
+											},
+										},
+									},
+									Command: []string{
+										"envoy",
+										"--concurrency",
+										"2",
+										"-c",
+										"/etc/envoy/envoy.yaml",
+									},
+									Resources: corev1.ResourceRequirements{
+										Requests: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("20m"),
+											corev1.ResourceMemory: resource.MustParse("20Mi"),
+										},
+										Limits: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("200m"),
+											corev1.ResourceMemory: resource.MustParse("300Mi"),
+										},
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{
+											Name:      "envoy-config",
+											MountPath: "/etc/envoy",
+										},
+										{
+											Name:      DeploymentName,
+											MountPath: "/etc/tls",
+										},
+									},
+								},
+							},
+							TerminationGracePeriodSeconds: pointer.Int64Ptr(30),
+							Volumes: []corev1.Volume{
+								{
+									Name: DeploymentName,
+									VolumeSource: corev1.VolumeSource{
+										Secret: &corev1.SecretVolumeSource{
+											SecretName: DeploymentName,
+										},
+									},
+								},
+								{
+									Name: VpnSeedServerTLSAuth,
+									VolumeSource: corev1.VolumeSource{
+										Secret: &corev1.SecretVolumeSource{
+											SecretName: VpnSeedServerTLSAuth,
+										},
+									},
+								},
+								{
+									Name: "vpn-seed-server-dh",
+									VolumeSource: corev1.VolumeSource{
+										Secret: &corev1.SecretVolumeSource{
+											SecretName: "vpn-seed-server-dh",
+										},
+									},
+								},
+								{
+									Name: "envoy-config",
+									VolumeSource: corev1.VolumeSource{
+										ConfigMap: &corev1.ConfigMapVolumeSource{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "vpn-seed-server-envoy-config",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			if nodeNetwork != nil {
+				deploy.Spec.Template.Spec.Containers[0].Env = append(deploy.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{Name: "NODE_NETWORK", Value: *nodeNetwork})
+			}
+			return deploy
+		}
+
+		destinationRule = func() *networkingv1beta1.DestinationRule {
```

Comment:
Hm, do we need this one to be a func? I assume it can be a regular var.

```suggestion
		destinationRule = &networkingv1beta1.DestinationRule{
```

Basically the same question applies for the below vars as well.

Reply:
Indeed, only for the deployment the function is required. However, the function do no harm as well and make it rather uniform.
So you are suggesting to change it for everything except for the deployment?

Reply:
> So you are suggesting to change it for everything except for the deployment?

Yes. At least I wouldn't use such func and rather user a simple var.

Reply:
Done with latest commit.